share/spack/spack-completion.bash

```diff
@@ -288,7 +288,7 @@ function _spack_debug {
     fi
 }
 
-function _spack_create-db-tarball {
+function _spack_create_db_tarball {
```

Comment:
@adamjstewart On rhel6 I get:
```
$ . share/spack/setup-env.sh 
-sh: `_spack_create-db-tarball': not a valid identifier
```
Honestly I didn't check how completion works. I thought it was a typo.